### PR TITLE
Bump to graphql-java 8.0

### PR DESCRIPTION
**Pull request by rwilliams-r7**
Upgrading graphql-java to version 8.0.  

Note junit had to be changed `EndToEndSpec.groovy`. Not 100% sure if this is correct, please review.

### CODE REVIEW

File: src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt [hunk 1]
```diff
@@ -142,7 +145,7 @@ internal class SchemaClassScanner(initialDictionary: BiMap<String, Class<*>>, al
         // Ensure all scalar definitions have implementations and add the definition to those.
         val scalars = scalarDefinitions.filter { !ScalarInfo.STANDARD_SCALAR_DEFINITIONS.containsKey(it.name) }.map { definition ->
             val provided = scalars[definition.name] ?: throw SchemaClassScannerError("Expected a user-defined GraphQL scalar type with name '${definition.name}' but found none!")
-            GraphQLScalarType(provided.name, SchemaParser.getDocumentation(definition) ?: provided.description, provided.coercing, definition)
+            GraphQLScalarType(provided.name, SchemaParser.getDocumentation(definition) ?: provided.description, provided.coercing, ArrayList<GraphQLDirective>(), definition)
```
Comment (rwilliams-r7) [inline src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt]: Added an emptyList. The new builder part of `GraphQLScalarType` did not allow `Coercing coercing` to be set. Have used the only constructor that allows me to pass the args required. With this in mind I have had to pass an empty list  `ArrayList<GraphQLDirective>()`, as it cannot be null.

File: src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy [hunk 2]
```diff
@@ -93,18 +93,16 @@ class EndToEndSpec extends Specification {
...
                 @Override
                 void onComplete() {
-
                 }
             })
             latch.await(3, TimeUnit.SECONDS)
 
         then:
-            returnedItem.id == 1
+            returnedItem.get("onItemCreated").id == 1
```
Comment (rwilliams-r7) [inline src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy]: Not 100% sure if this is correct.

File: src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt (hunk 1 above)
Comment (apottere) [reply src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt]: Would you mind changing this to `listOf()`?

File: src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy (hunk 2 above)
Comment (apottere) [reply src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy]: What prompted the change?

Comment (rwilliams-r7) [reply src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy]: `returnedItem.id` was null and of course the junit failed. Does it look correct?

Comment (rwilliams-r7) [reply src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy]: @apottere - are you happy enough with this?

**Comment by rwilliams-r7** [comment]
Can this be merged?

**Comment by maarek** [comment]
Any update on this?

**Comment by esovan** [comment]
Update? :)

**Comment by esovan** [comment]
Is there any way i can use this before it is merged to master? We really need it for our project. 😁

**Comment by filipncs** [comment]
@esovan If you already have an internal maven repository, you can simply fork the project and push the maven artifact to your repository. You can then use this artifact just like you used the official artifacts.

If none of that sounds familiar, you can get away with just building the library locally and including the jar in your project. But it's probably a good investment to figure out how to do your own library builds properly.

**Comment by esovan** [comment]
@filipncs Yes, i was planning on doing so if no other solution was suggested :) thanks though :)

File: src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy (hunk 2 above)
Comment (ryangardner) [reply src/test/groovy/com/coxautodev/graphql/tools/EndToEndSpec.groovy]: I dug into this - it looks like the reason the behavior change was due to this change in graphql-java: 

https://github.com/graphql-java/graphql-java/commit/adcd3f7e8248f7a591e2f8110ff0aa1910b341f9#diff-83f68d2746cff37c74d371d07db0e53b

"This commit adds the root field to subscription results to conform to
the current version of the GraphQL specification."

So it's an expected change to be spec compliant.

**Comment by apottere** [comment]
Sorry for the delay on this, it's syncing to central as `5.0.0` now.

**Pull request merged**